lhotse/dataset/speech_recognition.py

```diff
@@ -92,6 +93,10 @@ def __getitem__(self, cuts: CutSet) -> Dict[str, Union[torch.Tensor, List[str]]]
         """
         validate_for_asr(cuts)
 
+        if self.batch_counter > 0 and self.batch_counter % 100 == 0:
+            close_cached_file_handles()
```

Comment:
Shall we reset `self.batch_counter` in case it overflows if it runs for enough time?

Reply:
Python uses special integers that can never overflow, I believe.